codesign/bin/codesign.dart

```diff
@@ -44,58 +50,65 @@ Future<void> main(List<String> args) async {
         exit(1);
       }
     })
+    // Name of the certificate, e.g. FLUTTER.IO LLC
     ..addOption(
-      kCodesignCertName,
+      kCodesignCertNameOption,
       help: 'The name of the codesign certificate to be used when codesigning.',
     )
     ..addOption(
-      kAppSpecificPassword,
+      kAppSpecificPasswordOption,
       help: 'Unique password specifically for codesigning the given application.',
     )
     ..addOption(
-      kCodesignAppStoreId,
+      kCodesignAppStoreIdOption,
       help: 'Apple developer account email used for authentication with notary service.',
     )
     ..addOption(
-      kCodesignTeamId,
+      kCodesignTeamIdOption,
       help: 'Team-id is used by notary service for xcode version 13+.',
     )
     ..addOption(
-      kCommit,
+      kCommitOption,
       help: 'the Flutter engine commit revision for which Google Cloud Storage binary artifacts should be codesigned.',
     )
     ..addMultiOption(
-      kCodesignFilepath,
+      kCodesignFilepathOption,
       help: 'The zip file paths to be codesigned. Pass this option multiple'
           'times to codesign multiple zip files',
       valueHelp: 'darwin-x64/font-subset.zip',
     )
     ..addFlag(
-      kProduction,
+      kProductionFlag,
       help: 'whether we are going to upload the artifacts back to GCS for production',
     );
 
   final ArgResults argResults = parser.parse(args);
 
-  final Platform platform = LocalPlatform();
+  const Platform platform = LocalPlatform();
 
-  final String commit = getValueFromEnvOrArgs(kCommit, argResults, platform.environment)!;
-  final String codesignCertName = getValueFromEnvOrArgs(kCodesignCertName, argResults, platform.environment)!;
-  final String codesignUserName = getValueFromEnvOrArgs(kCodesignUserName, argResults, platform.environment)!;
-  final String appSpecificPassword = getValueFromEnvOrArgs(kAppSpecificPassword, argResults, platform.environment)!;
-  final String codesignAppstoreId = getValueFromEnvOrArgs(kCodesignAppStoreId, argResults, platform.environment)!;
-  final String codesignTeamId = getValueFromEnvOrArgs(kCodesignTeamId, argResults, platform.environment)!;
+  final String commit = getValueFromEnvOrArgs(kCommitOption, argResults, platform.environment)!;
```

Comment:
(No change needed).

Avoid passing parameters from env vars as it makes it harder to debug where values are getting from. It's easy to forget that the LUCI recipes configure the bots a certain way. For secrets it makes sense, and generally this is ok as a helper.

When you implement on the recipes side, this should only pass the cert via env vars.

Reply:
Very True!